cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -199,6 +201,36 @@ func generateInitramfsMounts() (err error) {
 	return nil
 }
 
+func canInstallAndRun(mst *initramfsMountsState) (bool, error) {
+	currentSeed, err := seed.Open(boot.InitramfsUbuntuSeedDir, mst.recoverySystem)
+	if err != nil {
+		return false, err
+	}
+	preseedSeed, ok := currentSeed.(seed.PreseedCapable)
+	if !ok {
+		return false, nil
+	}
+
+	if !preseedSeed.HasArtifact("preseed.tgz") {
+		return false, nil
+	}
+
+	// If kernel has fde-setup hook, then we should also have fde-setup in initramfs
+	kernelPath := filepath.Join(boot.InitramfsRunMntDir, "kernel")
+	gadgetHasFdeSetup := osutil.FileExists(filepath.Join(kernelPath, "meta", "hooks", "fde-setup"))
+	_, fdeSetupErr := exec.LookPath("fde-setup")
+	if gadgetHasFdeSetup && fdeSetupErr != nil {
+		return false, nil
+	}
+
+	gadgetPath := filepath.Join(boot.InitramfsRunMntDir, "gadget")
+	if osutil.FileExists(filepath.Join(gadgetPath, "meta", "hooks", "install-device")) {
+		return false, nil
+	}
+
+	return true, nil
```

Comment:
if we make this temporarily kernelHasFdeSetup then we can land this on master even without an implementation without  breaking edge for anything not yet setup for the feature

Reply:
Done in a717f2ce6bfc5a7376b0891e3a4757ef7a244170

Tests had to be tweaked.